File: src/services/deepSeekService.ts
import { api } from './api'

// DeepSeek 服务接口
export interface DeepSeekResponse {
  reply: string
}

// DeepSeek 服务
export const deepSeekService = {
  /**
   * 发送消息到 DeepSeek AI
   * @param message 用户消息
   * @param model 模型名称
   * @returns AI 回复
   */
  async chat(message: string, model: string = 'deepseek-chat'): Promise<DeepSeekResponse> {
    try {
      console.log(`发送消息到 DeepSeek AI，模型: ${model}，消息: ${message.substring(0, 50)}${message.length > 50 ? '...' : ''}`)

      try {
        // 尝试调用后端 API
        const response = await api.post<DeepSeekResponse>('/deepseek/chat', {
          message,
          model
        })

        console.log('DeepSeek AI 响应:', response)

        // 如果响应格式不正确，抛出错误
        if (!response || typeof response.reply !== 'string') {
          console.error('无效的 DeepSeek 响应格式:', response)
          throw new Error('无效的响应格式')
        }

        return response
      } catch (apiError: any) {
        // 如果是 401 未授权错误，返回模拟响应
        if (apiError.message && apiError.message.includes('401')) {
          console.warn('未授权错误，返回模拟响应')
          return {
            reply: `这是一个模拟响应。您发送的消息是: "${message}"。由于授权问题，无法连接到后端服务。`
          }
        }

        // 其他错误继续抛出
        throw apiError
      }
    } catch (error: any) {
      console.error('DeepSeek 聊天请求失败:', error)
      throw error
    }
  }
}

File: QuestionBankApi/DeepSeek/DeepSeekService.cs
/*
 * DeepSeekService.cs
 * ------------------
 * DeepSeek AI 服务实现类。
 * 负责封装与 DeepSeek API 的通信逻辑。
 */

using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;

namespace QuestionBankApi.DeepSeek;

/// <summary>
/// DeepSeek AI 服务，封装与 DeepSeek API 的通信
/// </summary>
public class DeepSeekService
{
    private readonly HttpClient _httpClient;
    private readonly string _apiKey;
    private readonly string _model;
    private readonly ILogger<DeepSeekService> _logger;

    /// <summary>
    /// 构造函数，注入配置和日志服务
    /// </summary>
    /// <param name="config">配置服务</param>
    /// <param name="logger">日志服务</param>
    public DeepSeekService(IConfiguration config, ILogger<DeepSeekService> logger)
    {
        _httpClient = new HttpClient();
        _logger = logger;

        // 从配置中获取 API 密钥和模型名称
        _apiKey = config.GetValue<string>("DeepSeek:ApiKey") ?? config["DeepSeek:ApiKey"];
        _model = config.GetValue<string>("DeepSeek:Model") ?? config["DeepSeek:Model"] ?? "deepseek-chat";

        // 记录密钥信息（注意不要记录完整密钥）
        if (string.IsNullOrEmpty(_apiKey))
        {
            _logger.LogWarning("未找到 DeepSeek API 密钥配置，将使用模拟响应");
            _apiKey = "sk-demo-key";
        }
        else
        {
            _logger.LogInformation("找到 DeepSeek API 密钥配置，密钥前缀: {Prefix}", _apiKey.Substring(0, Math.Min(8, _apiKey.Length)));
        }

        _logger.LogInformation("DeepSeek 服务已初始化，使用模型: {Model}", _model);
    }

    /// <summary>
    /// 与 DeepSeek AI 聊天
    /// </summary>
    /// <param name="userMessage">用户消息</param>
    /// <param name="model">模型名称（可选）</param>
    /// <returns>AI 回复</returns>
    public async Task<string> ChatAsync(string userMessage, string? model = null)
    {
        try
        {
            _logger.LogInformation("准备发送消息到 DeepSeek AI，模型: {Model}", model ?? _model);

            // 模拟响应，用于测试
            if (_apiKey == "sk-demo-key")
            {
                _logger.LogWarning("使用演示密钥，返回模拟响应");
                await Task.Delay(1000); // 模拟网络延迟
                return $"这是一个模拟响应。您发送的消息是: \"{userMessage}\"。请在 appsettings.json 中配置有效的 DeepSeek API 密钥。";
            }

            // 构建请求
            var url = "https://api.deepseek.com/v1/chat/completions";
            var requestBody = new
            {
                model = model ?? _model, // 优先使用参数，否则使用默认值
                messages = new[]
                {
                    new { role = "user", content = userMessage }
                }
            };

            // 序列化请求体
            var json = JsonSerializer.Serialize(requestBody);
            var request = new HttpRequestMessage(HttpMethod.Post, url);
            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", _apiKey);
            request.Content = new StringContent(json, Encoding.UTF8, "application/json");

            // 发送请求
            _logger.LogInformation("发送请求到 DeepSeek API");
            var response = await _httpClient.SendAsync(request);
            response.EnsureSuccessStatusCode();
            var responseString = await response.Content.ReadAsStringAsync();

            // 解析响应
            using var doc = JsonDocument.Parse(responseString);
            var content = doc.RootElement
                .GetProperty("choices")[0]
                .GetProperty("message")
                .GetProperty("content")
                .GetString();

            _logger.LogInformation("收到 DeepSeek AI 回复，长度: {Length} 字符", content?.Length ?? 0);
            return content ?? "无回复";
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "DeepSeek AI 聊天请求失败");
            throw;
        }
    }
}
